src/lib/types.hpp

```diff
@@ -206,6 +210,41 @@ std::pair<PredicateCondition, PredicateCondition> between_to_conditions(const Pr
 // Join, e.g., {GreaterThanEquals, LessThan} into BetweenUpperExclusive
 PredicateCondition conditions_to_between(const PredicateCondition lower, const PredicateCondition upper);
 
+enum class OperatorType {
```

Comment:
This enum class is quite large and, besides in `AbstractOperator`, we only need it for the memory resource. We do have some dedicated files for some (enum) types (cf. `file_type.hpp`). Maybe move this to `operators/operator_type.hpp`?

Reply:
Good idea!
But we would prefer to move it back to ´AbstractOperator´, but this causes some linker errors we have not fixed yet.